ui/shared/context/Notifications.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const NotificationsContext = React.createContext({});
+
+export class Notifications extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      notifications: [],
+      storedNotifications: JSON.parse(localStorage.getItem('notifications') || '[]'),
+    };
+    this.value = {
+      ...this.state,
+      notify: this.notify,
+      removeNotification: this.removeNotification,
+      clearStoredNotifications: this.clearStoredNotifications,
+    };
+  }
```

Comment:
I'm curious why these methods are part of the class member `this.value`. Seems like it might be cleaner to make it part of the state, and then pass the entire state to the provider?

Reply:
That's a good question.  I'll experiment with that.  This was an attempt to work around: https://reactjs.org/docs/context.html#caveats in a simple-ish way.  And avoid a call to ``setState`` from with ``componentDidMount`` after doing my ``bind`` calls.  But let me experiment a bit and see if there's a better way to achieve this.

Reply:
I couldn't find a good way to do this at this point.  Not without having to do a ``setState`` from ``componentDidMount``.  So I'll leave this as-is for now, but try your idea once we are fully a react app and I can but the ``bind`` calls into the constructor.